.github/workflows/mdspan-version-check.yml

```diff
@@ -0,0 +1,21 @@
+name: mdspan version check
+
+on:
+  workflow_call:
+
+permissions: read-all
+
+jobs:
+  mdspan-version-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        repository: kokkos/mdspan
+        ref: 28192956b30906a8c61948268f012d6ea1ce7c2a
+        path: kokkos-mdspan
+    - run: |
+        rm kokkos-mdspan/include/experimental/mdarray
```

Comment:
One thing to note is that while this will work, we *technically* may want to call an install since we use that as snapshot process (the install won't include the std-like headers by default). Don't think it's a huge deal right now as it amounts to the same thing

Reply:
By install you mean the result of `cmake --install ...` right ?

If this was the intended way to snapshot mdspan, we can update the workflow. The idea of the workflow is also to clarify the snapshot process of mdspan in Kokkos.